models/product.py

```diff
@@ -66,6 +67,23 @@ def one_or_none(result):
     raise MultipleLoadedResultsFound()
 
 
+@dataclass(frozen=True)
+class ProductGroupType:
+    slug: str
+    name: str
+    purchase_cls: type[Purchase]
+
+
+PRODUCT_GROUP_TYPES = [
```

Comment:
I'm muddling my way back through the design factors for the change: is there a reason these should go here instead of alongside the [similar-ish constants in `purchase.py`](https://github.com/emfcamp/Website/blob/f0b02a885fe2db4ae3045269dba9f61e6bd6ec8a/models/purchase.py#L12)?

Reply:
That list doesn't seem to be used at all, so it can probably be removed.

The root issue here is that there's inevitable duplication because we have a polymorphic type hierarchy under Purchase. Which of those Purchase types is created is determined by the Product's `ProductGroup.type`. And the point of #1214 (I haven't checked how relevant it still is) is that we should probably only be checking one of those.

Anyway I'm happy that this PR is a net improvement to the situation.